nitrocli/src/pinentry.rs

```diff
@@ -79,7 +79,8 @@ impl PinEntry {
 impl SecretEntry for PinEntry {
   fn cache_id(&self) -> Option<CowStr> {
     let model = self.model.to_string().to_lowercase();
-    let suffix = format!("{}:{}", model, self.serial);
+    // TODO: directly format serial instead?
+    let suffix = format!("{}:{:08x}", model, self.serial.as_u32());
```

Comment:
Should we use the new format with the `0x` prefix, or should we stick with the old format without prefix?

Reply:
I was going to look what we display on our `pinentry` dialog, but it seems you already did that.
> For consistency, we keep the old formatting for the cache ID used by
pinentry.

I have a slight preference for preserving the existing behavior.

Reply:
Actually, nevermind what I said. Given that we already use the other format in other places anyway I am okay with switching over. I'll take care of that.